gpio_test.go

```diff
@@ -0,0 +1,173 @@
+package gpio
+
+import (
+	"os"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+// validate and extract test setup
+func getEnv(t *testing.T) (gpioDev string, gpioIn uint32, gpioOut uint32, hasGpio bool, hasLoopback bool) {
+	gpioDev = os.Getenv("GPIO_TEST_DEV")
+	inStr := os.Getenv("GPIO_TEST_PIN")
+	outStr := os.Getenv("GPIO_TEST_PIN_LOOP")
+	if gpioDev == "" {
+		t.Skip("Please set GPIO_TEST_DEV, GPIO_TEST_PIN to run tests ")
+		return "", 0, 0, false, false
+	}
+	if inStr == "" {
+		t.Skip("Please set GPIO_TEST_DEV, GPIO_TEST_PIN to run tests ")
+	}
+	in, err := strconv.Atoi(inStr)
+	if err != nil {
+		t.Errorf("GPIO_TEST_PIN pin is unparsable: %s", err)
+		return "", 0, 0, false, false
+	} else {
+		gpioIn = uint32(in)
+	}
+	hasGpio = true
+	if outStr == "" {
+		t.Skip("Please set GPIO_TEST_PIN_LOOP and connect it with GPIO_TEST_PIN physically to run tests")
+	} else {
+		out, err := strconv.Atoi(outStr)
+		if err != nil {
+			t.Errorf("GPIO_TEST_PIN_LOOP is unparsable: %s", err)
+		} else {
+			gpioOut = uint32(out)
+			hasLoopback = true
+		}
+	}
+	if hasLoopback && gpioIn == gpioOut {
+		t.Errorf("test pins can't be same %d|%d", gpioIn, gpioOut)
+	}
+
+	return gpioDev, gpioIn, gpioOut, hasGpio, hasLoopback
+
+}
+
+func TestGPIO(t *testing.T) {
+	assert := assert.New(t)
+	testDevice, testPinInput, testPinOutput, hasGpio, hasLoopback := getEnv(t)
+	t.Logf("dev: %s in: %d, out: %d", testDevice, testPinInput, testPinOutput)
+	// no test pin provided, skip
+	if hasGpio {
+		return
+	}
+	chiper, err := Open(testDevice, "go-test")
+	if !assert.Nil(err, "Open should succeed") {
```

Comment:
There is `require.Nil,Equal,...` set of functions that act as SkipNow, Fatal to avoid 3 lines on if statement.

Reply:
I mean `require.NoError(err, "Open should succeed")` is equivalent to

```
if !assert.NoError(err, "Open should succeed") {
	t.FailNow()
}
```

But don't worry if you don't want. Current version is great.

Reply:
If I knew it existed I would've used it :) I've used goconvey before (mostly because of UI) but it kinda...fell apart when touching modules so I do not have much experience with other frameworks yet.